extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/exec/DataServerQueryHandler.java

```diff
@@ -300,16 +295,17 @@ private List<DataServerRequestDescriptor> createNextPendingRequests(
       if (segmentLoadInfo.getSegment().toDescriptor().equals(segmentDescriptorWithFullInterval)) {
         Set<DruidServerMetadata> servers = segmentLoadInfo.getServers()
                                                           .stream()
-                                                          .filter(druidServerMetadata -> includeSegmentSource.getUsedServerTypes()
-                                                                                                             .contains(druidServerMetadata.getType()))
+                                                          .filter(druidServerMetadata -> SegmentSource.REALTIME.getUsedServerTypes()
```

Comment:
To mirror the suggestion for `IndexerControllerContext`, consider moving this to a constant in `DartControllerContext`.